include/shady/runtime.h

```diff
@@ -27,8 +27,8 @@ const char* get_device_name(Device*);
 Device* get_an_device(Runtime*);
 
 Program* load_program(Runtime*, const char* program_src);
-Dispatch* launch_kernel(Program*, Device*, int dimx, int dimy, int dimz, int args_count, void** args);
-bool wait_completion(Dispatch*);
+Commands* launch_kernel(Program*, Device*, int dimx, int dimy, int dimz, int args_count, void** args);
```

Comment:
Why command_s_ ? I don't think the details of command buffers or anything like that should surface at this level, conceptually a kernel launch is just one command.

Reply:
I factored this out because we'll generally often need to record some commands and then wait for their completion. The code for that is pretty much always going to be the same. It's needed, for example, in the buffer copy stuff. In the process, I noticed that none of the `Dispatch`-specific stuff in `Dispatch` is ever actually used for anything, so I eventually just ended up changing `Dispatch` into a more generic `Commands`. But I agree, just calling it `Command` might be better API-wise. I'll change that.